### PR TITLE
dev-python/e4u: New Package

### PR DESCRIPTION
**Pull request by ConiKost**
dev-python/e4u: New Package

Closes: https://bugs.gentoo.org/652256

### CODE REVIEW

File: dev-python/e4u/e4u-0.1_rc4.ebuild [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit distutils-r1 eapi7-ver
+
+MY_PV="$(ver_rs 2 '' )"
```
Comment (mgorny) [inline dev-python/e4u/e4u-0.1_rc4.ebuild]: Better `${PV/_/}`.

File: dev-python/e4u/e4u-0.1_rc4.ebuild [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+DESCRIPTION="It's the bundle library for emoji4unicode"
```
Comment (mgorny) [inline dev-python/e4u/e4u-0.1_rc4.ebuild]: `it's` is redundant here.

File: dev-python/e4u/e4u-0.1_rc4.ebuild [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+HOMEPAGE="https://github.com/lambdalisue/e4u"
+SRC_URI="https://github.com/lambdalisue/e4u/archive/0.1rc4.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline dev-python/e4u/e4u-0.1_rc4.ebuild]: Use MY_PV? Also please name the tarball PN-MY_PV to match directory inside it.

File: dev-python/e4u/e4u-0.1_rc4.ebuild [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+LICENSE="GPL-3"
```
Comment (mgorny) [inline dev-python/e4u/e4u-0.1_rc4.ebuild]: Where did you find that?

File: dev-python/e4u/e4u-0.1_rc4.ebuild [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="dev-python/beautifulsoup:python-2[${PYTHON_USEDEP}]"
+
+DEPEND="${DEPEND}"
```
Comment (mgorny) [inline dev-python/e4u/e4u-0.1_rc4.ebuild]: Missing dependency of setuptools. Also check if it uses entry_points, in which setuptoots would be RDEP too.

File: dev-python/e4u/e4u-0.1_rc4.ebuild [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+S="${WORKDIR}/${PN}-${MY_PV}"
+
+DOCS=( README.rst )
```
Comment (mgorny) [inline dev-python/e4u/e4u-0.1_rc4.ebuild]: Isn't that redundant to the default?

File: dev-python/e4u/metadata.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+	<maintainer type="person">
+		<email>ck+gentoo@bl4ckb0x.de</email>
+		<name>Conrad Kostecki</name>
+	</maintainer>
+	<maintainer type="project">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<longdescription>
+		It's the bundle library for emoji4unicode.
```
Comment (mgorny) [inline dev-python/e4u/metadata.xml]: longdesc is only if there's longer version of the description.

File: dev-python/e4u/e4u-0.1_rc4.ebuild (hunk 2 above)
Comment (ConiKost) [reply dev-python/e4u/e4u-0.1_rc4.ebuild]: I rewrote it now.

File: dev-python/e4u/e4u-0.1_rc4.ebuild (hunk 5 above)
Comment (ConiKost) [reply dev-python/e4u/e4u-0.1_rc4.ebuild]: I am not sure, if I understand it, but I tend to say yes after looking.

File: dev-python/e4u/e4u-0.1_rc4.ebuild (hunk 6 above)
Comment (ConiKost) [reply dev-python/e4u/e4u-0.1_rc4.ebuild]: Hm yes. Looking into it, I had `README.md` in my mind, but it's actually `README*` Dropping it.

File: dev-python/e4u/metadata.xml (hunk 7 above)
Comment (ConiKost) [reply dev-python/e4u/metadata.xml]: I added a more longer one.

File: dev-python/e4u/e4u-0.1_rc4.ebuild [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+DESCRIPTION="A library for handling unicode emoji and carrier's emoji"
+HOMEPAGE="https://github.com/lambdalisue/e4u"
+SRC_URI="https://github.com/lambdalisue/e4u/archive/${MY_PV}.tar.gz -> ${PN}-${MY_PV}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="dev-python/beautifulsoup:python-2[${PYTHON_USEDEP}]
+	dev-python/setuptools"
```
Comment (mgorny) [inline dev-python/e4u/e4u-0.1_rc4.ebuild]: PYTHON_USEDEP!111111oneoneoneelven

Comment (ConiKost) [reply dev-python/e4u/e4u-0.1_rc4.ebuild]: I could swear, I had done this :P Sorry, fixed now. I need that vacation now x)

**Comment by ConiKost** [comment]
> Sorry that I haven't noticed this earlier. Run the tests, please.

What do you mean? I don't see any requested changes?
Repoman didn't complained about missing PYTHON_USEDEP?

**Comment by mgorny** [comment]
I mean the package's test suite.

**Comment by ConiKost** [comment]
> I mean the package's test suite.

Okay, I managed to fix those tests. The problem was, that they used an url for the xml, which was hosted on google code, which is now dead. After fixing this and fixing the init for tests, they work fine.

BUT: Tests won't work with FEATURES="network-sandbox", as they rely on quering specific servers for the emoji. I don't think, this can be changed. What could be done in this case? I currently check this and throw a die.

**Comment by mgorny** [comment]
Upstream-wise, mocking would probably work.

Downstream-wise, just add `RESTRICT="test"` with appropriate explanation in comment and people who want network access can unrestrict them.

**Comment by ConiKost** [comment]
Thanks, added now.

**Comment by ConiKost** [comment]
Sorry, that I didn't saw your reply. I did it now and it seems to work fine.

**Comment by mgorny** [comment]
Thanks!

**Pull request closed without merging**